node/task.ts

```diff
@@ -243,41 +248,110 @@ export function loc(key: string, ...param: any[]): string {
 // Input Helpers
 //-----------------------------------------------------
 /**
- * Gets a variables value which is defined on the build definition or set at runtime.
+ * Gets a variable value that is defined on the build definition or set at runtime.
  * 
  * @param     name     name of the variable to get
  * @returns   string
  */
 export function getVariable(name: string): string {
     var varval;
-    if (vault) {
-        varval = vault.retrieveSecret('SECRET_' + name.replace(/\./g, '_').toUpperCase());
+
+    // get the metadata
+    var info: VariableInfo;
+    var key: string = getVariableKey(name);
+    if (variableInfo.hasOwnProperty(key)) {
+        info = variableInfo[key];
     }
 
-    if (!varval) {
-        varval = process.env[name.replace(/\./g, '_').toUpperCase()];
+    if (info && info.secret) {
+        // get the secret value
+        varval = vault.retrieveSecret('SECRET_' + key);
+    } else {
+        // get the public value
+        varval = process.env[key];
+
+        // fallback for pre 2.104.1 agent
+        if (!varval && name.toUpperCase() == 'AGENT.JOBSTATUS') {
+            varval = process.env['agent.jobstatus'];
+        }
     }
 
     debug(name + '=' + varval);
     return varval;
 }
 
 /**
- * Sets a variables which will be available to subsequent tasks as well.
+ * Gets all variables that are defined on the build definition or set at runtime.
+ * Requires a 2.104.1 agent or higher.
```

Comment:
The function requires a 2.104.1 agent to be full featured, but may still be useful on a pre-2.104.1 agent. Let's chat about it, I also don't want to overcomplicate it.